server/api/project/project.controller.js

```diff
@@ -76,3 +79,36 @@ exports.destroy = function(req, res) {
 function handleError(res, err) {
   return res.send(500, err);
 }
+
+exports.upload = function(req, res) {
+  var form = new multiparty.Form();
+  form.parse(req, function(err, fields, files) {
+    console.log(files.file[0].path);
+    var file = files.file[0];
+    var name = file.path.substring(file.path.lastIndexOf('/')).substring(1);
+    var cwd = process.cwd();
+    var path = cwd + '/server/static/uploads/' + req.params.username + '/' + req.params.project;
+    var destPath = path + '/' + name;
+    if(!fs.existsSync(path)){
+      mkdirp.sync(path);
+    }
+    fs.rename(file.path, destPath, function (err) {
+      if (err) console.error(err)
```

Comment:
These should return errors to the clients. Console errors on the server are just going to be eaten in the logs and never seen